main.tf

```
@@ -65,3 +76,26 @@ resource "aws_s3_bucket_public_access_block" "public_access_block" {
   restrict_public_buckets = true
 }
 
+resource "aws_s3_bucket_inventory" "inventory" {
+  count = var.enable_bucket_inventory ? 1 : 0
+
+  bucket = aws_s3_bucket.private_bucket.id
+  name   = "WeeklyInventory"
```

This probably doesn't matter all that much but will help with clarity, but since the frequency in configurable
```suggestion
  name   = "BucketInventory"
```